cmd/util/api.go

```diff
@@ -34,13 +35,24 @@ func GetAPIClientV2(cmd *cobra.Command, cfg types.Bacalhau) (clientv2.API, error
 	}
 	base := fmt.Sprintf("%s://%s:%d", apiSheme, apiHost, apiPort)
 
+	metastore := repo.NewMetadataStore(cfg.DataDir)
+	installationID, err := metastore.ReadInstallationID()
+	if err != nil {
+		return nil, err
+	}
+	instanceID, err := metastore.ReadInstanceID()
+	if err != nil {
+		return nil, err
+	}
 	bv := version.Get()
 	headers := map[string][]string{
-		apimodels.HTTPHeaderBacalhauGitVersion: {bv.GitVersion},
-		apimodels.HTTPHeaderBacalhauGitCommit:  {bv.GitCommit},
-		apimodels.HTTPHeaderBacalhauBuildDate:  {bv.BuildDate.UTC().String()},
-		apimodels.HTTPHeaderBacalhauBuildOS:    {bv.GOOS},
-		apimodels.HTTPHeaderBacalhauArch:       {bv.GOARCH},
+		apimodels.HTTPHeaderBacalhauGitVersion:          {bv.GitVersion},
+		apimodels.HTTPHeaderBacalhauGitCommit:           {bv.GitCommit},
+		apimodels.HTTPHeaderBacalhauBuildDate:           {bv.BuildDate.UTC().String()},
+		apimodels.HTTPHeaderBacalhauBuildOS:             {bv.GOOS},
+		apimodels.HTTPHeaderBacalhauArch:                {bv.GOARCH},
+		apimodels.HTTPHeaderBacalhauBuildInstallationID: {installationID},
+		apimodels.HTTPHeaderBacalhauInstanceID:          {instanceID},
```

Comment:
should be `HTTPHeaderBacalhauInstallationID` instead of `HTTPHeaderBacalhauBuildInstallationID`. Also should we always send the headers or only if they have values?

Reply:
will only include if value is present: https://github.com/bacalhau-project/bacalhau/pull/4450#discussion_r1767521569